src/main/java/org/myrobotlab/framework/repo/IvyWrapper.java

```diff
@@ -490,34 +507,43 @@ synchronized public void install(String location, String[] serviceTypes) {
         save();
       }
 
-      ArtifactDownloadReport[] artifacts = report.getAllArtifactsReports();
-      for (int i = 0; i < artifacts.length; ++i) {
-        ArtifactDownloadReport ar = artifacts[i];
-        Artifact artifact = ar.getArtifact();
-        // String filename = IvyPatternHelper.substitute("[originalname].[ext]",
-        // artifact);
-
-        File file = ar.getLocalFile();
-        String filename = file.getAbsoluteFile().getAbsolutePath();
-        log.info("{}", filename);
+      if (report == null) {
+        String errorDetail = String.format("there was problems resolving dependencies %s", (Object[]) serviceTypes); 
+        log.error(errorDetail);
+        publishStatus(Status.newInstance(Repo.class.getSimpleName(), StatusLevel.ERROR, Repo.INSTALL_FINISHED,
+            errorDetail));
+        throw new RuntimeException(errorDetail);
+      } else {
 
-        if ("zip".equalsIgnoreCase(artifact.getExt())) {
-          info("unzipping %s", filename);
-          try {
-            Zip.unzip(filename, "./");
-            info("unzipped %s", filename);
-          } catch (Exception e) {
-            log.error("unable to unzip file {}", filename, e);
+        ArtifactDownloadReport[] artifacts = report.getAllArtifactsReports();
+        for (int i = 0; i < artifacts.length; ++i) {
+          ArtifactDownloadReport ar = artifacts[i];
+          Artifact artifact = ar.getArtifact();
+          // String filename =
+          // IvyPatternHelper.substitute("[originalname].[ext]",
+          // artifact);
+
+          File file = ar.getLocalFile();
+          String filename = file.getAbsoluteFile().getAbsolutePath();
+          log.info("{}", filename);
+
+          if ("zip".equalsIgnoreCase(artifact.getExt())) {
+            info("unzipping %s", filename);
+            try {
+              Zip.unzip(filename, "./");
+              info("unzipped %s", filename);
+            } catch (Exception e) {
+              log.error("unable to unzip file {}", filename, e);
+            }
           }
         }
-      }
-
-      publishStatus(Status.newInstance(Repo.class.getSimpleName(), StatusLevel.INFO, Repo.INSTALL_FINISHED, String.format("finished install of %s", (Object[]) serviceTypes)));
 
-    } catch (Exception e) {
-      error(e.getMessage());
-      log.error(e.getMessage(), e);
-    }
+        publishStatus(Status.newInstance(Repo.class.getSimpleName(), StatusLevel.INFO, Repo.INSTALL_FINISHED, String.format("finished install of %s", (Object[]) serviceTypes)));
+      }
+//    } catch (Exception e) {

```

Comment:
Like above, I think catching and providing a helpful error message would be good here, like `delete ivy cache`. Then we can also terminate the program with `System.exit()` or rethrow.